bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/handler/OpenWebNetThermoregulationHandler.java

```diff
@@ -0,0 +1,375 @@
+            refreshDevice(false);
+        }
+    }
+
+    @Override
+    protected void handleChannelCommand(ChannelUID channel, Command command) {
+        switch (channel.getId()) {
+            case CHANNEL_TEMP_SETPOINT:
+                handleSetpointCommand(command);
+                break;
+            case CHANNEL_FUNCTION:
+                handleFunction(command);
+                break;
+            case CHANNEL_MODE:
+                handleMode(command);
+                break;
+            case CHANNEL_FAN_SPEED:
+                handleSetFanSpeedCommand(command);
+                break;
+            default: {
+                logger.warn("handleChannelCommand() Unsupported ChannelUID {}", channel.getId());
+            }
+        }
+    }
+
+    @Override
+    protected void requestChannelState(ChannelUID channel) {
+        logger.debug("requestChannelState() thingUID={} channel={}", thing.getUID(), channel.getId());
+        refreshDevice(false);
+    }
+
+    @Override
+    protected Where buildBusWhere(String wStr) throws IllegalArgumentException {
+        WhereThermo wt = new WhereThermo(wStr);
+        if (wt.isProbe()) {
+            isTempSensor = true;
+        }
+        return wt;
+    }
+
+    @Override
+    protected String ownIdPrefix() {
+        return Who.THERMOREGULATION.value().toString();
+    }
+
+    private void handleSetFanSpeedCommand(Command command) {
+        logger.debug("handleSetFanSpeedCommand() (command={})", command);
+        if (command instanceof StringType) {
+            Where w = deviceWhere;
+            if (w != null) {
+                try {
+                    FAN_COIL_SPEED speed = FAN_COIL_SPEED.valueOf(command.toString());
+                    send(Thermoregulation.requestWriteFanCoilSpeed(w.value(), speed));
+                } catch (OWNException e) {
+                    logger.warn("handleSetFanSpeedCommand() {}", e.getMessage());
+                } catch (IllegalArgumentException e) {
+                    logger.warn("handleSetFanSpeedCommand() Unsupported command {} for thing {}", command,
+                            getThing().getUID());
+                    return;
+                }
+            }
+        } else {
+            logger.warn("handleSetFanSpeedCommand() Unsupported command {} for thing {}", command, getThing().getUID());
+        }
+    }
+
+    private void handleSetpointCommand(Command command) {
+        logger.debug("handleSetpointCommand() (command={})", command);
+        if (command instanceof QuantityType || command instanceof DecimalType) {
+            Where w = deviceWhere;
+            if (w != null) {
+                BigDecimal value = BigDecimal.ZERO;
+                if (command instanceof QuantityType) {
+                    Unit<Temperature> unit = CELSIUS;
+                    QuantityType<Temperature> quantity = commandToQuantityType(command, unit);
+                    value = quantity.toBigDecimal();
```

Comment:
You need to convert the temperature value with `quantity.toUnit()` into the unit you/openwebnet expects it. The user can also send °F temperatures.

Reply:
fixed